### ISSUE TITLE
Python 3.10 compatibility

### ISSUE THREAD

**Issue by cp2004** [eyal0/OctoPrint-PrintTimeGenius#269]
**Describe the bug**
```
2021-11-02 15:49:47,968 - py.warnings - WARNING - /home/pi/oprint/lib/python3.7/site-packages/octoprint_PrintTimeGenius/__init__.py:186: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated
```
In Python 3.10, this stopped working, which means that the line that calls it stops working as well.

To retain compatibility with Python 2.7, it needs to be wrapped up to cover both cases, example below:

https://github.com/OctoPrint/OctoPrint/blob/945825f03fbd2417a009dd33daf9f787aff60211/src/octoprint/util/__init__.py#L29-L33

**To Reproduce**
1. Read/run code, view deprecation warnings. Haven't tested what breaks, but I know in Python 3.10 it will break.

**Expected behavior**
No breaking 🙂 

**OctoPrint Version:** 1.7.1

**PrintTimeGenius Version (if you know):** Latest, checked in the default branch on GH.

This is probably low priority, but we have seen a few people installing Py3.10 (for example on Windows) and it breaks plugins. OctoPrint has full compatibility to Py 3.10 since 1.7.0 (in theory!) if you want to test against it.

**Comment by eyal0** [comment]
Would you like to send a PR?

**Comment by cp2004** [comment]
I can't test it, which is why I was hesitant to send a PR

**Comment by eyal0** [comment]
Ah.  Well, when I get around to it, I can do it.  I'd use pyenv I think.

**Comment by LMBernardo** [comment]
The changes in my fork as of commit ecb210a9fb58536aea70e24a20f408355ba70637 are functional for me with Python 3.10. If someone wants to test this on lower versions of Python I can do a PR.

**Comment by Nathan22211** [comment]
cp2004, just as a side note. Octoprint dropped support for python 2.7 after 1.8 I'm pretty sure

**Comment by maludwig** [comment]
Quick little PR to fix this, tested with pyenv, on python 3.2, 3.3, 3.9, and 3.10

https://github.com/eyal0/OctoPrint-PrintTimeGenius/pull/308

**Comment by eyal0** [comment]
Fixed in https://github.com/eyal0/OctoPrint-PrintTimeGenius/pull/308
